R/pkg/R/DataFrame.R

```diff
@@ -677,25 +677,44 @@ setMethod("unique",
 #' collect(sample(df, TRUE, 0.5))
 #'}
 setMethod("sample",
-          # TODO : Figure out how to send integer as java.lang.Long to JVM so
-          # we can send seed as an argument through callJMethod
           signature(x = "DataFrame", withReplacement = "logical",
-                    fraction = "numeric"),
-          function(x, withReplacement, fraction) {
+                    fraction = "numeric", seed = "missing"),
+          function(x, withReplacement, fraction, seed) {
             if (fraction < 0.0) stop(cat("Negative fraction value:", fraction))
             sdf <- callJMethod(x@sdf, "sample", withReplacement, fraction)
             dataFrame(sdf)
           })
 
 #' @rdname sample
+#' @name sample
+setMethod("sample",
+          # we can send seed as an argument through callJMethod
+          signature(x = "DataFrame", withReplacement = "logical",
+                    fraction = "numeric", seed = "numeric"),
```

Comment:
you could in fact merge these overload/variant into one. Please see this for an example:
https://github.com/apache/spark/blob/5011f264fb53705c528250bd055acbc2eca2baaa/R/pkg/R/DataFrame.R#L1177